qa-tests-backend/src/test/groovy/DiagnosticBundleTest.groovy

```diff
@@ -101,11 +107,31 @@ class DiagnosticBundleTest extends BaseSpecification {
         }
 
         def response = given()
-                .config(RestAssuredConfig.newConfig()
-                    .sslConfig(SSLConfig.sslConfig().relaxedHTTPSValidation().allowAllHostnames()))
-                .headers(headers)
-                .when()
-                .get("https://${Env.mustGetHostname()}:${Env.mustGetPort()}/api/extensions/diagnostics")
+            .config(RestAssured.config()
+                .httpClient(HttpClientConfig.httpClientConfig()
+                    .httpClientFactory(new HttpClientFactory() {
+            @Override
+            HttpClient createHttpClient() {
+                HttpClient httpClient = new SystemDefaultHttpClient()
+                // Retry 10 times on [InterruptedIOException, UnknownHostException, ConnectException,
+                // SSLException].
+                httpClient.setHttpRequestRetryHandler(new StandardHttpRequestRetryHandler(10, true))
```

Comment:
Alternatively, we can use `withRetry` helper here:
https://github.com/stackrox/stackrox/blob/cc9a05c29aada8187cf7d4c38446c39d05e1aed4/qa-tests-backend/src/main/groovy/services/ImageService.groovy#L68-L75

It would allow us to control interval with which we retry this call

Reply:
In general I think low level failures should be retried on the transport level, i.e. http / grpc clients with a retry handler. That way you can also ensure only retry-able errors are actually retried.

However, you are probably right that in this case it's better to be consistent with the rest of the codebase. Also I guess having fewer Java factories is a plus.